lib/feed.xml

```diff
@@ -53,26 +53,12 @@
       <content type="html" xml:base="{{ post.url | prepend: url_base | xml_escape }}">{{ post.content | strip | xml_escape }}</content>
 
       {% assign post_author = post.author | default: post.authors[0] | default: site.author %}
-      {% assign post_author_email = false %}
-      {% assign post_author_uri = false %}
+      {% assign post_author = site.data.authors[post_author] | default: post_author %}
```

Comment:
I'm on the road so I can't write out a full example, but just checking if the string is a key of the authors object and has a value, otherwise assume the string is a name.